### PR TITLE
Release v0.9.0

### PR DESCRIPTION
**Pull request by benashz**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,31 @@
+## 0.9.0 (October 8th, 2024)
+
+Features:
+* Add support for syncing [HVS rotating secrets](https://developer.hashicorp.com/hcp/docs/vault-secrets/auto-rotation): [GH-893](https://github.com/hashicorp/vault-secrets-operator/pull/893)
+* Add support for syncing [HVS dynamic secrets](https://developer.hashicorp.com/hcp/docs/vault-secrets/dynamic-secrets): [GH-917](https://github.com/hashicorp/vault-secrets-operator/pull/917) [GH-939](https://github.com/hashicorp/vault-secrets-operator/pull/939) [GH-934](https://github.com/hashicorp/vault-secrets-operator/pull/934) [GH-941](https://github.com/hashicorp/vault-secrets-operator/pull/941)
```
Comment (tvoran) [inline CHANGELOG.md:5]: Should we mention https://github.com/hashicorp/vault-secrets-operator/pull/889 in here somewhere?

**Review by tvoran — APPROVED** [review]
Should we update the go version for this release? Looks like it's on 1.22.2 right now.

File: CHANGELOG.md (hunk 1 above)
Comment (benashz) [reply CHANGELOG.md]: Added in d7fb9b38ba476b1ac7db65efb8be6bb23857428b

**Review by stevealmyHC — COMMENTED** [review]
Feature list looks accurate, for the HVS private beta, announced in public docs.

**Comment by benashz** [comment]
> Should we update the go version for this release? Looks like it's on 1.22.2 right now.

Bumped in fe8d23c30279feb065b03267ba38d4e0d39c8d40

**Pull request merged**